filters/imagefilter.go

```diff
@@ -43,65 +41,74 @@ func init() {
 
 type ImageFilter interface {
 	CreateOptions(image *bimg.Image) (*bimg.Options, error)
+	CanBeMerged(other *bimg.Options, self *bimg.Options) bool
+	Merge(other *bimg.Options, self *bimg.Options) *bimg.Options
 }
 
 func HandleImageResponse(ctx filters.FilterContext, f ImageFilter) {
-	rsp := ctx.Response()
-
-	defer rsp.Body.Close()
-
-	rsp.Header.Del("Content-Length")
-
-	r, err := handleImageTransform(rsp.Body, f)
+	image := ctx.StateBag()[SkropImage].(*bimg.Image)
 
+	opt, err := f.CreateOptions(image)
```

Comment:
Question: is this scope executed when the setupResponse() filter fails and sets an error response? I think it should be explicitly prevented. How shunting works is that when you call ctx.Serve() the remaining response filters are still executed.